lib/business_time/time_extensions.rb

```diff
@@ -196,5 +189,14 @@ def business_time_until(to_time, *currency)
     def during_business_hours?(*currency)
       self.workday?(*currency) && self.to_i.between?(Time.beginning_of_workday(self).to_i, Time.end_of_workday(self).to_i)
     end
+
+    def workday_considering_holidays?(holidays)
+      stringified = strftime("%F")
+      if holidays.all? { |h| h.is_a?(String) }
+        !holidays.include?(stringified)
+      else
+        !holidays.any? { |h| h.to_date == self.to_date }
```

Comment:
Yes, actually this ` holidays.all?` call is a de-optimization to support non Strings in `BusinessTime::Config.currency_holidays`. I would prefer allowing only one type/format for both `BusinessTime::Config.holidays` and `BusinessTime::Config.currency_holidays`.

active-suppport related:
* For some reason I didn't found `none?` in Enumerable and automatically though that was part of active-support, I'll change it, thanks.
* I'll check if active-support `Time#change` whould be ok with this `yyyy_mm_dd`, thanks.